modules/phase_field/src/postprocessors/FeatureFloodCount.C

```diff
@@ -587,12 +572,11 @@ FeatureFloodCount::mergeSets(bool use_periodic_boundary_info)
   for (processor_id_type rank = 0; rank < n_procs; ++rank)
     for (unsigned int map_num = 0; map_num < _maps_size; ++map_num)
     {
-      for (std::vector<FeatureData>::iterator it = _partial_feature_sets[rank][map_num].begin();
-           it != _partial_feature_sets[rank][map_num].end(); ++it)
+      for (auto & feature : _partial_feature_sets[rank][map_num])
       {
-        if (!it->_merged)
+        if (!feature._merged)
         {
-          _feature_sets[map_num].push_back(MooseSharedPointer<FeatureData>(new FeatureData(*it)));
+          _feature_sets[map_num].push_back(std::unique_ptr<FeatureData>(new FeatureData(feature)));
```

Comment:
Right - it's already an rvalue... so it'll call the version of `push_back()` that takes an rvalue reference and ultimately do an rvalue assignment (calling the move operator).

That said - I _do_ think we should get used to using `emplace_back()`.  It can be a good optimization.  I actually have a `push_back`/`emplace_back` in a critical section in my new code where I'm creating shared pointers and putting them on the end of a `std::vector` and `emplace_back()` gave a measurable speed improvement (in my case I have a good upper bound for the number of things in that vector so I'm able to do a good `reserve()` up front to eliminate any speed loss due to re-allocation).